File: work/DSA/javascript/recursion_1/sumOfArray.js
let n = 3
let input = [9, 8, 9]

function sum(input, n) {
    if (n === 0) {
        return 0;
    }
    return sum(input.slice(1), n - 1) + input[0]
}

let ans = sum(input, n)
console.log(ans)

File: work/DSA/c++/recursion_1/sumOfArray.cpp
#include <bits/stdc++.h>
using namespace std;

int sum(int input[], int n)
{
    if (n == 1)
    {
        return input[0];
    }
    return sum(input + 1, n - 1) + input[0];
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;
    int *input = new int[n];
    for (int i = 0; i < n; i++)
    {
        cin >> input[i];
    }
    cout << sum(input, n) << endl;
}